### PR TITLE
google asr

### PR DESCRIPTION
**Pull request by brentspell**
These changes add the google asr as a speech pipeline component. Speech recognition is triggered when the context becomes active and completed when the context becomes inactive again.

### CODE REVIEW

File: src/main/java/com/pylon/spokestack/google/GoogleSpeechRecognizer.java [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+        if (context.isActive() && this.request == null)
+            begin(context);
+        else if (!context.isActive() && this.request != null)
+            commit();
+        else if (context.isActive())
+            send(frame);
+    }
+
+    private void begin(SpeechContext context) {
+        BidiStreamingCallable<
```
Comment (noelweichbrodt) [inline src/main/java/com/pylon/spokestack/google/GoogleSpeechRecognizer.java]: *Bidi* really?

File: src/main/java/com/pylon/spokestack/google/GoogleSpeechRecognizer.java [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+        ResponseObserver(SpeechContext speechContext) {
+            this.context = speechContext;
+        }
+
+        @Override
+        public void onNext(StreamingRecognizeResponse message) {
+            // we aren't using intermediate recognition results,
+            // so just attach the first result
+            StreamingRecognitionResult result = message.getResults(0);
+            SpeechRecognitionAlternative alt = result.getAlternatives(0);
```
Comment (noelweichbrodt) [inline src/main/java/com/pylon/spokestack/google/GoogleSpeechRecognizer.java]: oh java

File: src/test/java/com/pylon/spokestack/google/GoogleSpeechRecognizerTest.java [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+
+        public ApiStreamObserver getRequests() {
+            return this.requests;
+        }
+
+        public ApiStreamObserver getResponses() {
+            ArgumentCaptor<ApiStreamObserver> captor =
+                ArgumentCaptor.forClass(ApiStreamObserver.class);
+            verify(this.callable)
+                .bidiStreamingCall(captor.capture());
```
Comment (noelweichbrodt) [inline src/test/java/com/pylon/spokestack/google/GoogleSpeechRecognizerTest.java]: actual lol

File: src/test/java/com/pylon/spokestack/google/GoogleSpeechRecognizerTest.java [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+
+        // responses
+        client.getResponses().onNext(StreamingRecognizeResponse.newBuilder()
+            .addResults(StreamingRecognitionResult.newBuilder()
+                .addAlternatives(SpeechRecognitionAlternative.newBuilder()
+                    .setTranscript("test")
+                    .setConfidence((float)0.75)
+                    .build())
+                .build())
+            .build()
```
Comment (noelweichbrodt) [inline src/test/java/com/pylon/spokestack/google/GoogleSpeechRecognizerTest.java]: https://www.forbes.com/sites/outofasia/2018/02/28/dutertes-ambitious-build-build-build-project-to-transform-the-philippines-could-become-his-legacy/#6ea70a071a7f

File: src/test/java/com/pylon/spokestack/google/GoogleSpeechRecognizerTest.java (hunk 3 above)
Comment (brentspell) [reply src/test/java/com/pylon/spokestack/google/GoogleSpeechRecognizerTest.java]: I'm not sure who is the captor and who is the captee at this point.

**Comment by brentspell** [comment]
[android speech framework](https://trello.com/c/xmIajshe/26-android-speech-framework)

**Pull request merged**